miio/yeelight.py

```diff
@@ -312,63 +331,163 @@ def on(self, transition=0, mode=0):
         4: color flow
         5: moonlight
         """
-        if transition > 0 or mode > 0:
-            return self.send("set_power", ["on", "smooth", transition, mode])
-        return self.send("set_power", ["on"])
+        if (
+            lamp == YeelightSubLightType.Background
+            and not self.model_specs.has_background_light
+        ):
+            raise YeelightException("Background lamp not supported")
+
+        miio_command = SUBLIGHT_PROP_PREFIX[lamp] + "set_power"
+        miio_param = ["on"]
+        if transition > 0:
+            miio_param += ["smooth", transition]
+        else:
+            miio_param += ["sudden", 0]
+        if mode > 0:
+            if (
+                lamp == YeelightSubLightType.Main
+                and not self.model_specs.supports_color
+            ):
+                if mode == 2:
+                    raise YeelightException("RGB color not supported")
+                if mode == 3:
+                    raise YeelightException("HSV color not supported")
+            miio_param += [mode]
+        return self.send(miio_command, miio_param)
 
     @command(
         click.option("--transition", type=int, required=False, default=0),
+        click.option(
+            "--lamp", type=int, required=False, default=YeelightSubLightType.Main
+        ),
         default_output=format_output("Powering off"),
     )
-    def off(self, transition=0):
+    def off(self, transition=0, lamp=YeelightSubLightType.Main):
         """Power off."""
+        if (
+            lamp == YeelightSubLightType.Background
+            and not self.model_specs.has_background_light
+        ):
+            raise YeelightException("Background lamp not supported")
```

Comment:
This is used often enough that it warrants its own private check method, e.g., calling `self._error_on_non_background(lamp)` should suffice.